server.go

```diff
@@ -3289,7 +3290,7 @@ func setupRPCListeners() ([]net.Listener, error) {
 // newServer returns a new dcrd server configured to listen on addr for the
 // decred network type specified by chainParams.  Use start to begin accepting
 // connections from peers.
-func newServer(ctx context.Context, listenAddrs []string, db database.DB, chainParams *chaincfg.Params, dataDir string) (*server, error) {
+func newServer(ctx context.Context, cancel context.CancelFunc, listenAddrs []string, db database.DB, chainParams *chaincfg.Params, dataDir string) (*server, error) {
```

Comment:
As mentioned elsewhere, this change to take the cancel func needs to be reverted.